src/main/java/spoon/template/Substitution.java

```diff
@@ -114,9 +116,20 @@ private Substitution() {
 		final Map<String, Object> extendedParams = new HashMap<String, Object>(templateParameters);
 		extendedParams.put(templateOfType.getSimpleName(), typeRef);
 		List<CtType<?>> generated = (List) new SubstitutionVisitor(f, extendedParams).substitute(templateOfType.clone());
+
+		CompilationUnit cu = f.createCompilationUnit();
+		cu.setDeclaredPackage(targetPackage);
+		cu.setDeclaredTypes(generated);
+
 		for (CtType<?> ctType : generated) {
 			targetPackage.addType(ctType);
+			if (ctType.getPosition() != SourcePosition.NOPOSITION) {
+				SourcePosition olderPosition = ctType.getPosition();
+				SourcePosition sourcePosition = f.createSourcePosition(cu, olderPosition.getSourceStart(), olderPosition.getSourceEnd(), null);
+				ctType.setPosition(sourcePosition);
```

Comment:
I think that source positions of generated type are not related to source position of template file. The generated file should have "undefined" source start/end positions. Or what is the use case behind the copying of source positions. I afraid that such invalid source positions might only confuse the pretty printer, if it is in mode "keep lines".

And if any code needs a CompilationUnit, then that code should create it. I think it is not the responsibility of type generation code.

Reply:
> I think that source positions of generated type are not related to source position of template file. 

I agree, I kept this because it was the previous behaviour: the position was just a clone of the sourceposition of the template. 

> The generated file should have "undefined" source start/end positions. Or what is the use case behind the copying of source positions.

I don't know if we have some rules in Spoon for such case: @monperrus @tdurieux any idea?

Reply:
For the moment all the new elements have a SourcePosition.NOPOSITION position

Reply:
> For the moment all the new elements have a SourcePosition.NOPOSITION position

Yes but if we want to put a SourcePosition with a compilation unit, but without specifying the positions, is there a recommended practice?

Reply:
My first thought is to move out the compilation unit from the position.
But I don't really like the idea, I don't see a good practice.

Reply:
Using "-1" for positions?...

Reply:
It was like that before and it is not a good specification

Reply:
So what about a `PartialSourcePosition`?